test/erlfmt_SUITE.erl

```diff
@@ -1154,25 +1155,34 @@ snapshot_insert_pragma_with(Config) when is_list(Config) ->
     snapshot_same("pragma.erl", [{pragma, insert} | Config]).
 
 snapshot_same(Module, Config) ->
-    DataDir = ?config(data_dir, Config),
     Pragma = proplists:get_value(pragma, Config, ignore),
-    {ok, OriginalBin} = file:read_file(filename:join(DataDir, Module)),
-    Original = unicode:characters_to_list(OriginalBin),
-    case erlfmt:format_string(Original, [{pragma, Pragma}]) of
-        {ok, Original, _} -> ok;
-        {skip, _} -> ok;
-        {ok, Other, _} -> ct:fail({unexpected, Other, expected, Original});
-        Other -> ct:fail("unexpected: ~p~n", [Other])
-    end.
+    snapshot_match(Module, Module, Config, [{pragma, Pragma}]).
 
 snapshot_formatted(Module, Config) ->
+    % Sanity check via idempotence: the reference file is up-to-date.
     snapshot_same(Module ++ ".formatted", Config),
+    snapshot_match(Module ++ ".formatted", Module, Config, []).
+
+snapshot_match(FormattedModule, Module, Config, Options) ->
     DataDir = ?config(data_dir, Config),
-    {ok, FormattedBin} = file:read_file(filename:join([DataDir, Module ++ ".formatted"])),
+    {ok, FormattedBin} = file:read_file(filename:join([DataDir, FormattedModule])),
     Formatted = unicode:characters_to_list(FormattedBin),
     {ok, OriginalBin} = file:read_file(filename:join([DataDir, Module])),
     Original = unicode:characters_to_list(OriginalBin),
-    ?assertMatch({ok, Formatted, _}, erlfmt:format_string(Original, [])).
+    case erlfmt:format_string(Original, Options) of
+        {ok, Formatted, _} -> ok;
```

Comment:
Maybe this should be `{ok, Original, _}` ?

Reply:
Why? You want to check the result of format_string against the reference file.
I think 'Expected' would be less ambiguous, but I kept the existing terminology to have the minimum diff possible!

Reply:
I was looking at original code and not paying much attention.
I probably missed something.

![Screen Shot 2021-05-17 at 5 08 15 PM](https://user-images.githubusercontent.com/1829933/118521007-7db20a80-b732-11eb-836f-86a2092bb84b.png)

Reply:
The code you are quoting is for `snapshot_same` where both the reference and the file to format are the same.
This is now expressed by passing `Module` twice to the helper function.

Reply:
Okay great, that makes sense.  My bad.